contrib/guix/libexec/build.sh

```diff
@@ -230,6 +230,10 @@ case "$HOST" in
     *mingw*)  HOST_LDFLAGS="-Wl,--no-insert-timestamp" ;;
 esac
 
+case "$HOST" in
+    x86_64-linux-gnu)  HARDENED_LDFLAGS="-Wl,-z,cet-report=error" ;;
```

Comment:
I don't think splitting flags into `LDFLAGS` and `HARDENED_LDFLAGS` inside Guix makes sense, as, everything in Guix is meant to be a "hardened" build.

Reply:
This is done to prevent disruption of internal flag checks.

Reply:
```
< snip >
x86_64-linux-gnu-ld: /tmp/ccDAU1BA.o: error: missing IBT and SHSTK properties
collect2: error: ld returned 1 exit status
E
======================================================================
ERROR: test_ELF (__main__.TestSecurityChecks)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/distsrc-base/distsrc-059bcf847da0-x86_64-linux-gnu/./contrib/devtools/test-security-check.py", line 67, in test_ELF
    self.assertEqual(call_security_check(cxx, source, executable, pass_flags + ['-fcf-protection=none']), (1, executable + ': failed CONTROL_FLOW'))
  File "/distsrc-base/distsrc-059bcf847da0-x86_64-linux-gnu/./contrib/devtools/test-security-check.py", line 42, in call_security_check
    subprocess.run([*cxx,source,'-o',executable] + env_flags() + options, check=True)
  File "/gnu/store/fn4rrnx7vx1bl6mccrhm95y6br8yg9s6-python-minimal-3.10.7/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
< snip >
```

Reply:
```
x86_64-linux-gnu-ld: /tmp/ccwqGS4u.o: error: missing IBT and SHSTK properties
collect2: error: ld returned 1 exit status
E
======================================================================
ERROR: test_ELF (__main__.TestSecurityChecks)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/distsrc-base/distsrc-5fda052191b0-x86_64-linux-gnu/./contrib/devtools/test-security-check.py", line 67, in test_ELF
    self.assertEqual(call_security_check(cxx, source, executable, pass_flags + ['-fcf-protection=none']), (1, executable + ': failed CONTROL_FLOW'))
  File "/distsrc-base/distsrc-5fda052191b0-x86_64-linux-gnu/./contrib/devtools/test-security-check.py", line 42, in call_security_check
    subprocess.run([*cxx,source,'-o',executable] + env_flags() + options, check=True)
  File "/gnu/store/fn4rrnx7vx1bl6mccrhm95y6br8yg9s6-python-minimal-3.10.7/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
```

Reply:
GIven that the whole point of these scripts is to sanity check the (release) tests, using the release flags, and in the Guix env; I don't think an undocumented workaround, of just hiding certain release flags, from the test scripts, is the right thing to do.

Reply:
OK. What can you suggest?

Reply:
> OK. What can you suggest?

My suggestion is to remove the changes that have been added to facilitate bypassing tests (i.e separating the `LDFLAGS` into `LDFLAGS` and `HARDENED_LDFLAGS`, so that some flags are skipped when tests are performed). If the build environment has changed such that the current set of tests don't pass with the new changes, then the tests should be reworked, not skipped.